baybe/surrogates/custom.py

```diff
@@ -7,12 +7,11 @@
 
 It is planned to solve this issue in the future.
 """
+from __future__ import annotations
```

Comment:
I dont immediately understand why this has become needed in several places where you've added it, can you elaborate?

Reply:
Because we don't want to use string types for `Tensor` which is imported only for type checking and not runtime.
For example, we should write `def _posterior(self, candidates: Tensor) -> Tuple[Tensor, Tensor]` like:
`def _posterior(self, candidates: "Tensor") -> Tuple["Tensor", "Tensor"]`

Reply:
and why was that not a problem before hence requiring the change?